src/utils/dom.js

```diff
@@ -127,24 +127,27 @@ export function scrollIntoView(itemEl: ?HTMLElement, options?: Object = {}): voi
 
 /**
  *
- * A React hook that tells you if an element (passed in as a ref) has content that overflows its container,
+ * A React hook that returns a ref function to be used to measure if referenced content is overflowing its container.
  * i.e., if the text is wider than the box around it.
  *
- * @param {{ current: null | typeof undefined | HTMLElement }} contentRef
- * @return {boolean}
+ * @param {void}
+ * @return {{isRefOverflowed: boolean, ref: Function}}
  */
-export function useIsContentOverflowed(contentRef: { current: null | typeof undefined | HTMLElement }): boolean {
-    const [isContentOverflowed, setIsContentOverflowed] = React.useState<boolean>(false);
-    const { current } = contentRef;
```

Comment:
I believe the previous implementation would run after each render (correctly) if this destructure was moved inside the hook and it was not passed as a dep to the function

Reply:
The function should also `setIsContentOverflowed(false);` to reset when the condition inverts, which it's not currently doing